frontend/src/components/commons/SideNavigation.jsx

```diff
@@ -37,16 +37,29 @@ class SideNavigation extends Component {
   };
 
   state = {
-    currentNode: this.props.currentNode,
-    currentBody: this.props.navSpecs[this.props.currentNode].body
+    currentNode: this.props.currentNode
   };
 
   changeNode(id) {
-    this.setState({ currentNode: id, currentBody: this.props.navSpecs[id].body });
+    this.setState({ currentNode: id });
+    this.refs[id].scrollIntoView({ block: "start", behavior: "smooth" });
   }
 
+  onScroll = () => {
+    const curVw = document.getElementById("side-nav-grid-content-cell").getBoundingClientRect();
```

Comment:
variables like this make this code hard to read - can we spell out all these variables?

Reply:
renamed in 14c36aa